scheduler/util.go

```diff
@@ -732,3 +732,80 @@ func updateNonTerminalAllocsToLost(plan *structs.Plan, tainted map[string]*struc
 		}
 	}
 }
+
+// newAllocUpdateFn is a factory for the scheduler to create an allocUpdateType
+// function for the reconciler
+func newAllocUpdateFn(ctx Context, stack Stack, evalID string) allocUpdateType {
```

Comment:
The name of this function and doc string is confusing. Better as `genericAllocUpdateFn` and describe its behavior a bit more.